jenkins/Jenkinsfile-build

```diff
@@ -0,0 +1,11 @@
+library('private-pipeline-library')
+library('jenkins-shared')
+
+mavenSnapshotPipeline(performSonarAnalysis: true,
```

Comment:
Or at least may not want most of the rules we use on Nexus.

Reply:
actually we do sonar analysis on this one - 
![image](https://user-images.githubusercontent.com/5182904/41053442-7e54b4dc-6989-11e8-9059-88205fd842ca.png)

I was kinda surprised since it's a fork. But, I figured I'd default to parity with the bamboo build first (and this provided an excuse to actually implement sonar analysis in the jenkins-shared library).

Reply:
> actually we do sonar analysis on this one -

Ah, I checked the build and overlooked it I guess.